gillespy2/core/model.py

```diff
@@ -935,9 +935,20 @@ def run(self, solver=None, timeout=0, t=None, increment=None, show_labels=True,
             from gillespy2.core import log
             log.warning('show_labels = False is deprecated. Future releases '
                         'of GillesPy2 may not support this feature.')
+
         if t is None:
             t = self.tspan[-1]
 
+        if increment is not None and self.timespan is not None:
```

Comment:
This always evaluates to True if an increment is provided, because a `timespan` is constructed for us if none is provided in the model class constructor, [found here](https://github.com/StochSS/GillesPy2/blob/5ffe1897612ba61609a683a4d87123b09ad96c87/gillespy2/core/model.py#L164). As a result, using an increment will _always_ raise an error.

To replicate this behavior:

```python
class ModelEx(gillespy2.Model):
    def __init__(self):
        gillespy2.Model.__init__(self, name="ModelEx")
        s = gillespy2.Species(name="s", initial_value=0)
        k = gillespy2.Parameter(name="k", expression=1.0)
        r = gillespy2.Reaction(name="r", products={s: 1}, rate=k)
        self.add_species(s)
        self.add_parameter(k)
        self.add_reaction(r)
        # Notice that no timespan is set explicitly

#%%

model = ModelEx()
# Despite not explicitly setting a timespan, the tspan property exists.
# This will cause an error to be raised.
print(model.tspan is None)
# -> False

#%%

# This code raises an error: `Both increment and timespan are set`
# Even though we only supplied an increment ourselves, the timespan was created for us.
# As a result, passing an increment will always fail.
model.run(solver=gillespy2.NumPySSASolver, increment=0.1)

```

Reply:
This is fixed in the latest commits